cmd/packaging/darwin-pkg.go

```diff
@@ -1,86 +1,23 @@
-
-	createDockerfile(packagingFormat, []string{
+// DarwinPkgPackagingTask packaging for darwin as pkg
+var DarwinPkgPackagingTask = &packagingTask{
+	packagingFormatName: "darwin-pkg",
+	dependsOn: map[*packagingTask]string{
+		DarwinBundlePackagingTask: "flat/root/Applications",
+	},
+	templateFiles: map[string]string{
+		"darwin-pkg/PackageInfo.tmpl.tmpl":  "flat/base.pkg/PackageInfo.tmpl",
+		"darwin-pkg/Distribution.tmpl.tmpl": "flat/Distribution.tmpl",
+	},
+	dockerfileContent: []string{
 		"FROM ubuntu:bionic",
 		"RUN apt-get update && apt-get install cpio git make g++ wget libxml2-dev libssl1.0-dev zlib1g-dev -y",
 		"WORKDIR /tmp",
 		"RUN git clone https://github.com/hogliux/bomutils && cd bomutils && make > /dev/null && make install > /dev/null",
 		"RUN wget https://storage.googleapis.com/google-code-archive-downloads/v2/code.google.com/xar/xar-1.5.2.tar.gz && tar -zxvf xar-1.5.2.tar.gz > /dev/null && cd xar-1.5.2 && ./configure > /dev/null && make > /dev/null && make install > /dev/null",
-	})
-
-	printInitFinished(packagingFormat)
-}
-
-func BuildDarwinPkg(buildVersion string) {
-	log.Infof("Building darwin-bundle first")
-	BuildDarwinBundle(buildVersion)
-	projectName := pubspec.GetPubSpec().Name
-	packagingFormat := "darwin-pkg"
-	tmpPath := getTemporaryBuildDirectory(projectName, packagingFormat)
-	defer func() {
-		err := os.RemoveAll(tmpPath)
-		if err != nil {
-			log.Errorf("Could not remove temporary build directory: %v", err)
-			os.Exit(1)
-		}
-	}()
-	log.Infof("Packaging pkg in %s", tmpPath)
-
-	err := copy.Copy(build.OutputDirectoryPath("darwin-bundle"), filepath.Join(tmpPath, "flat", "root", "Applications"))
-	if err != nil {
-		log.Errorf("Could not copy build folder: %v", err)
-		os.Exit(1)
-	}
-	fileutils.CopyTemplateDir(packagingFormatPath(packagingFormat), filepath.Join(tmpPath), getTemplateData(projectName, buildVersion))
-	outputFileName := projectName + "-" + buildVersion + ".pkg"
-	runDockerPackaging(tmpPath, packagingFormat, []string{
-		"(cd flat/root && find . | cpio -o --format odc --owner 0:80 | gzip -c ) > flat/base.pkg/Payload",
-		"&&", "mkbom -u 0 -g 80 flat/root flat/base.pkg/Bom",
-		"&&", "(cd flat && xar --compression none -cf '../" + outputFileName + "' * )",
-	})
-
-	outputFilePath := filepath.Join(build.OutputDirectoryPath("darwin-pkg"), outputFileName)
-	err = copy.Copy(filepath.Join(tmpPath, outputFileName), outputFilePath)
-	if err != nil {
-		log.Errorf("Could not move pkg: %v", err)
-		os.Exit(1)
-	}
-
-	printPackagingFinished(packagingFormat)
+	},
+	packagingCommand:    "(cd flat/root && find . | cpio -o --format odc --owner 0:80 | gzip -c ) > flat/base.pkg/Payload && mkbom -u 0 -g 80 flat/root flat/base.pkg/Bom && (cd flat && xar --compression none -cf '../{{.projectName}}-{{.version}}.pkg' * )",
```

Comment:
`packagingCommand` is really a bash script, not a single command. Maybe rename to `packagingScriptTemplate`?